core/templates/pages/classroom-admin-page/classroom-admin-page.component.ts

```diff
@@ -0,0 +1,346 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Classroom admin component.
+ */
+
+import { Component, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { ClassroomBackendApiService, ClassroomBackendDict, ClassroomDict } from '../../domain/classroom/classroom-backend-api.service';
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { AppConstants } from 'app.constants';
+import { ClassroomEditorConfirmModalComponent } from './modals/classroom-editor-confirm-modal.component';
+import { DeleteClassroomConfirmModalComponent } from './modals/delete-classroom-confirm-modal.component';
+import { CreateNewClassroomModalComponent } from './modals/create-new-classroom-modal.component';
+import { EditableTopicBackendApiService } from 'domain/topic/editable-topic-backend-api.service';
+import cloneDeep from 'lodash/cloneDeep';
+
+@Component({
+  selector: 'oppia-classroom-admin-page',
+  templateUrl: './classroom-admin-page.component.html',
+})
+export class ClassroomAdminPageComponent implements OnInit {
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion. For more information, see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  constructor(
+    private classroomBackendApiService: ClassroomBackendApiService,
+    private editableTopicBackendApiService: EditableTopicBackendApiService,
+    private ngbModal: NgbModal,
+  ) {}
+
+  classroomCount: number = 0;
+  classroomIdToClassroomName: {[classroomId: string]: string} = {};
+  selectedClassroomDict!: ClassroomDict;
+  updatedClassroomDict!: ClassroomDict;
+
+  classroomId: string = '';
+  classroomName: string = '';
+  urlFragment: string = '';
+  courseDetails: string = '';
+  topicListIntro: string = '';
+  topicIds: string[] = [];
+  existingClassroomNames: string[] = [];
+  topicIdToPrerequisiteTopicIds: {[topicId: string]: string[]} = {};
+
+  pageIsInitialized: boolean = false;
+  classroomDataIsChanged: boolean = false;
+  classroomDetailsIsShown: boolean = false;
+  classroomViewerMode: boolean = false;
+  classroomEditorMode: boolean = false;
+  savingClassroomData: boolean = false;
+
+  classroomNameExceedsMaxLen: boolean = false;
```

Comment:
```suggestion
  classroomNameIsTooLong: boolean = false;
```

Ditto for URL fragment

Reply:
Updated